trio/_socket.py

```diff
@@ -178,9 +225,10 @@ def numeric_only_failure(exc):
             # idna.encode will error out if the hostname has Capital Letters
             # in it; with uts46=True it will lowercase them instead.
             host = _idna.encode(host, uts46=True)
-    hr = _resolver.get(None)
+    hr: HostnameResolver | None = _resolver.get(None)
```

Comment:
I think these type hints should also be inferred?

Reply:
yep, removing it has no effect. Though idk if having redundant explicit type hints is a bad thing.